interpreter/cling/lib/Interpreter/LookupHelper.cpp

```diff
@@ -52,6 +53,7 @@ namespace cling {
                                 llvm::StringRef code,
                                 llvm::StringRef bufferName,
                                 LookupHelper::DiagSetting diagOnOff) {
+    static std::map<llvm::hash_code, FileID> codeCache;
```

Comment:
We should remove this static variable here. Why not making `prepareForParsing` a private member of the `LookupHelper` and move codeCache there, too.

Reply:
Is static making the code slower? I'm not sure why you want to remove static.

Reply:
Static makes the cache global. It should be local to the LookupHelper object.